engine/gamesys/src/gamesys/resources/res_collision_object.cpp

```diff
@@ -173,51 +177,61 @@ namespace dmGameSystem
                     {
                         resource->m_TileGridResource = (TileGridResource*)res;
                         resource->m_TileGrid = 1;
-                        // Add the tile grid as the first and only shape
+
                         dmArray<dmPhysics::HCollisionShape2D>& shapes = resource->m_TileGridResource->m_GridShapes;
                         uint32_t shape_count = shapes.Size();
-                        if (shape_count > COLLISION_OBJECT_MAX_SHAPES)
-                        {
-                            dmLogWarning("The collision object '%s' has a tile map containing more than %d layers, the rest will be ignored.", filename, COLLISION_OBJECT_MAX_SHAPES);
-                            shape_count = COLLISION_OBJECT_MAX_SHAPES;
-                        }
+                        uint32_t total_shapes_count = embedded_shape_count + shape_count;
+                        resource->m_Shapes2D.SetCapacity(total_shapes_count);
+                        resource->m_ShapeTranslation.SetCapacity(total_shapes_count);
+                        resource->m_ShapeRotation.SetCapacity(total_shapes_count);
+                        resource->m_Shapes2D.SetSize(total_shapes_count);
+                        resource->m_ShapeTranslation.SetSize(total_shapes_count);
+                        resource->m_ShapeRotation.SetSize(total_shapes_count);
```

Comment:
Thinking about this a bit more, there's really no need to have these as dynamic arrays.
We allocate them once, that's it.
I recommend replacing them with C arrays instead.